### ISSUE TITLE
Comparison with mermaid stateDiagram

### ISSUE THREAD

**Issue by forivall** [sverweij/state-machine-cat#126]
from a quick glance, it's unclear if state-machine-cat is a superset of mermaid-js. It would be nice, if so, if it could parse mermaidjs state diagrams as a language.

### Context

it would just be neat, especially since i'm used to mermaid's syntax

### Proposed Behavior
it would be a new input type

### Current Behavior
supported input are smcat, json and scxml

### Considered alternatives
well, just using mermaid, of course.

**Comment by sverweij** [comment]
Hi @forivall state-machine-cat is not a super set of mermaid, syntax wise. 

I hadn't looked into mermaid for a few years and wasn't even aware they had state machine support right now, so thanks for the heads-up. 

Feature wise state-machine-cat currently likely is a superset of mermaid; from a quick glance mermaid doesn't seem to support history states, terminations, junctions, activities, conditions, actions and events yet. It does seem to have some stuff that influence layout (right_of, left_of), though, and that's something smc does not have - and it might have a more flexible/ better render engine (not tried yet).

That said transforming mermaid syntax into state-machine-cat's internal format ([json schema here](https://github.com/sverweij/state-machine-cat/blob/develop/src/parse/smcat-ast.schema.json)) should be doable. A PR welcome.

**Comment by forivall** [comment]
Yeah I definitely was thinking superset, feature-wise. Deffo not syntax. 👍
